### PR TITLE
python3-Levenshtein: update to 0.23.0

### LINKED ISSUE #46545: Ulauncher broken after python 3.12 upgrade
### Is this a new report?

Yes

### System Info

Void 6.5.6_1 x86_64

### Package(s) Affected

ulauncher-5.15.4_1, python3-Levenshtein-0.12.0_9

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

_No response_

### Expected behaviour

ulauncher command should open application popup

### Actual behaviour

Command fails with python undefined symbol error due to **PyUnicode_AS_UNICODE** remotion on python 3.12
Affected package is python3-Levenshtein

```py
Traceback (most recent call last):
  File "/usr/bin/ulauncher", line 27, in <module>
    from ulauncher.main import main
  File "/usr/lib/python3.12/site-packages/ulauncher/main.py", line 25, in <module>
    from ulauncher.ui.windows.UlauncherWindow import UlauncherWindow
  File "/usr/lib/python3.12/site-packages/ulauncher/ui/windows/UlauncherWindow.py", line 23, in <module>
    from ulauncher.search.Search import Search
  File "/usr/lib/python3.12/site-packages/ulauncher/search/Search.py", line 3, in <module>
    from ulauncher.api.server.ExtensionSearchMode import ExtensionSearchMode
  File "/usr/lib/python3.12/site-packages/ulauncher/api/server/ExtensionSearchMode.py", line 3, in <module>
    from ulauncher.api.server.DeferredResultRenderer import DeferredResultRenderer
  File "/usr/lib/python3.12/site-packages/ulauncher/api/server/DeferredResultRenderer.py", line 11, in <module>
    from ulauncher.api.shared.item.ResultItem import ResultItem
  File "/usr/lib/python3.12/site-packages/ulauncher/api/shared/item/ResultItem.py", line 5, in <module>
    from ulauncher.utils.text_highlighter import highlight_text
  File "/usr/lib/python3.12/site-packages/ulauncher/utils/text_highlighter.py", line 1, in <module>
    from ulauncher.utils.fuzzy_search import get_matching_indexes
  File "/usr/lib/python3.12/site-packages/ulauncher/utils/fuzzy_search.py", line 4, in <module>
    from Levenshtein import distance
  File "/usr/lib/python3.12/site-packages/Levenshtein/__init__.py", line 1, in <module>
    from Levenshtein import _levenshtein
ImportError: /usr/lib/python3.12/site-packages/Levenshtein/_levenshtein.so: undefined symbol: PyUnicode_AS_UNICODE
```

### Steps to reproduce

1. Open terminal
2. execute command ulauncher

### LINKED ISSUE #46581: python3-Levenshtein broken with Python 3.12
This links successfully against symbols removed from Python 3.12, but the linker fails to find these symbols at runtime when the module is loaded. The package is used by `python3-thefuzz` and `ulauncher`.

### PR DESCRIPTION
**Pull request by chrysos349**
fixes #46581
fixes #46545

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl x
  - armv7l x
  - armv6l-musl x

### CODE REVIEW

File: srcpkgs/python3-hatch-fancy-pypi-readme/template
Comment (classabbyamp) [inline srcpkgs/python3-hatch-fancy-pypi-readme/template:1]: this package is useless anywhere but for uploading to pypi, [patch it out instead](https://github.com/void-linux/void-packages/blob/master/srcpkgs/python3-jsonschema/patches/0001-patch-out-hatch-fancy-pypi-readme.patch)

File: srcpkgs/python3-hatch-vcs/template
Comment (classabbyamp) [inline srcpkgs/python3-hatch-vcs/template:1]: already exists as `hatch-vcs`

File: srcpkgs/python3-rapidfuzz/template [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+checksum=448d031d9960fea7826d42bd4284156fc68d3b55a6946eb34ca5c6acf960577b
+
+pre_build() {
+	# ugly hack to help cmake find python3 headers
+	if [ "$CROSS_BUILD" ]; then
+		rm -rf ${XBPS_BUILDDIR}/../${py3_inc}
+		ln -sf ${XBPS_BUILDDIR}/../${XBPS_CROSS_BASE}/${py3_inc} \
+			${XBPS_BUILDDIR}/../${py3_inc}
+	fi
+}
```
Comment (classabbyamp) [inline srcpkgs/python3-rapidfuzz/template:24]: don't do this, there should be a cmake flag that can be passed to tell it where things are.

`zap` and re-`binary-bootstrap` your masterdirs

File: srcpkgs/python3-hatch-vcs/template
Comment (chrysos349) [reply srcpkgs/python3-hatch-vcs/template:1]: Saw it when the check was done.

File: srcpkgs/python3-hatch-fancy-pypi-readme/template
Comment (chrysos349) [reply srcpkgs/python3-hatch-fancy-pypi-readme/template:1]: Will do.

File: srcpkgs/python3-rapidfuzz/template (hunk 1 above)
Comment (chrysos349) [reply srcpkgs/python3-rapidfuzz/template]: @classabbyamp done

**Comment by ahesford** [comment]
Drop the README.voidlinux on RapidFuzz. There's not a lot of value in a two-line file that says "you can use this with numpy", and the optional dependency isn't really a specific consideration of Void. Users of RapidFuzz should be expected to understand how to use it with or without optional dependencies.

**Comment by chrysos349** [comment]
@ahesford dropped.

**Comment by chrysos349** [comment]
updated `python3-rapidfuzz` to `3.4.0`

**Pull request merged**